jcl-custom-rules/src/main/java/com/sonarsource/jcl/example/checks/MyCustomRule.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2021-2024 SonarSource SA
+ * All rights reserved
+ * mailto:info AT sonarsource DOT com
+ */
+package com.sonarsource.jcl.example.checks;
+
+import org.sonar.plugins.jcl.api.checks.InitContext;
+import org.sonar.plugins.jcl.api.checks.JclCheck;
+import org.sonar.plugins.jcl.api.tree.statement.ExecStatementTree;
+import org.sonar.plugins.jcl.api.tree.statement.JobStatementTree;
+import org.sonar.check.Rule;
+
+@Rule(key = MyCustomRule.RULE_KEY)
+public class MyCustomRule implements JclCheck {
+
+  public static final String RULE_KEY = "Rule1";
+
+  @Override
+  public void init(InitContext context) {
+    context.register(ExecStatementTree.class, (ctx, exec) -> {
+      if (exec.name() != null && !exec.name().value().startsWith("STEP")) {
```

Comment:
Minor remark: The corresponding code in `sonar-jcl` defines a variable `SyntaxToken name` and reuses it twice.
I think we should keep the code aligned and either apply the inlining in `sonar-jcl` or extract the `SyntaxToken name` here too.